src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/ClaimsByHearingVenueExcelReportCreationService.java

```diff
@@ -0,0 +1,231 @@
+        styleForHeaderCell.setFont(getFont(workbook));
+        createCell(rowReportTitle, 0, reportPrintedOnDescription, styleForHeaderCell);
+    }
+
+    private void constructCaseExcelRow(XSSFWorkbook workbook, XSSFSheet sheet, int rowIndex,
+                                       ClaimsByHearingVenueReportDetail item) {
+        XSSFRow row = sheet.createRow(rowIndex);
+        row.setHeight((short)(row.getHeight() * 4));
+        int columnIndex = 0;
+        var cellStyle = getCellStyle(workbook);
+        createCell(row, columnIndex, item.getCaseReference(), cellStyle);
+        createCell(row, columnIndex + 1, item.getDateOfReceipt(), cellStyle);
+        createCell(row, columnIndex + 2, item.getClaimantPostcode(), cellStyle);
+        createCell(row, columnIndex + 3, item.getClaimantWorkPostcode(), cellStyle);
+        createCell(row, columnIndex + 4, item.getRespondentPostcode(), cellStyle);
+        createCell(row, columnIndex + 5, item.getRespondentET3Postcode(), cellStyle);
+    }
+
+    private Font getFont(XSSFWorkbook workbook) {
+        Font font = workbook.createFont();
+        font.setBold(true);
+        font.setFontName("Calibre");
+        font.setColor(IndexedColors.DARK_GREEN.getIndex());
+        font.setFontHeightInPoints((short)16);
+        return font;
+    }
+
+    private CellStyle getCellStyle(XSSFWorkbook workbook) {
+        CellStyle cellStyle = workbook.createCellStyle();
+        cellStyle.setVerticalAlignment(VerticalAlignment.CENTER);
+        var font = getFont(workbook);
+        font.setColor(IndexedColors.BLACK1.getIndex());
+        font.setFontHeightInPoints((short)14);
+        font.setBold(false);
+        cellStyle.setFont(font);
+        cellStyle.setFillForegroundColor(IndexedColors.WHITE1.getIndex());
+        cellStyle.setFillPattern(FillPatternType.SOLID_FOREGROUND);
+        cellStyle.setAlignment(HorizontalAlignment.CENTER);
+        cellStyle.setBorderTop(BorderStyle.NONE);
+        cellStyle.setBorderLeft(BorderStyle.NONE);
+        cellStyle.setBorderRight(BorderStyle.NONE);
+        cellStyle.setBorderBottom(BorderStyle.THIN);
+        cellStyle.setBottomBorderColor(IndexedColors.GREY_25_PERCENT.getIndex());
+        return cellStyle;
+    }
+
+    private static byte[] writeExcelFileToByteArray(XSSFWorkbook workbook) {
+        var bos = new ByteArrayOutputStream();
+        try {
+            workbook.write(bos);
+            workbook.close();
+        } catch (IOException e) {
+            log.error("Error generating the excel report");
+            throw new ReportException("Error generating the excel report", e);
+        }
+
+        return bos.toByteArray();
+    }
+
+    private void createCell(XSSFRow row, int cellIndex, String value, CellStyle style) {
+        Cell cell = row.createCell(cellIndex);
+        cell.setCellStyle(style);
+
+        if (!Strings.isNullOrEmpty(value) && !value.isBlank()) {
```

Comment:
You can replace these two with StringUtils.isNotBlank
I prefer Apache StringUtils library to google Strings because the Javadoc provides examples